cellengine/resources/gate.py

```diff
@@ -1,124 +1,109 @@
 from __future__ import annotations
-import attr
 import importlib
-from typing import Dict, List, Optional
 from math import pi
-from custom_inherit import doc_inherit
+from operator import itemgetter
+from typing import Any, Dict, List, Optional
+
+from attr import define, field
+from numpy import array, mean, stack
 
 import cellengine as ce
-from cellengine.utils.helpers import get_args_as_kwargs
-from cellengine.payloads.gate import _Gate
-from cellengine.payloads.gate_utils import (
-    format_rectangle_gate,
-    format_split_gate,
-    format_polygon_gate,
-    format_ellipse_gate,
-    format_quadrant_gate,
-    format_range_gate,
+from cellengine.resources.fcs_file import FcsFile
+from cellengine.utils import parse_fcs_file_args
+from cellengine.utils import converter, generate_id, readonly
+from cellengine.utils.helpers import (
+    get_args_as_kwargs,
+    normalize,
+    remove_keys_with_none_values,
 )
+from collections import Mapping
```

Comment:
In the incoming push I'm changing this to:
```suggestion
import sys
if sys.version_info[:2] >= (3, 8):
    from collections.abc import Mapping, MutableMapping
else:
    from collections import Mapping, MutableMapping
```
for newer Python support.